test/sys/test_signal.rs

```diff
@@ -1,7 +1,39 @@
+use libc;
+use nix;
 use nix::unistd::*;
 use nix::sys::signal::*;
 
 #[test]
 fn test_kill_none() {
     kill(getpid(), None).ok().expect("Should be able to send signal to myself.");
 }
+
+#[test]
+fn test_signal() {
+    let _m = ::SIGNAL_MTX.lock().expect("Mutex got poisoned by another test");
+    unsafe {
+        signal(Signal::SIGINT, SigHandler::SigIgn).unwrap();
+        raise(Signal::SIGINT).unwrap();
+        assert_eq!(signal(Signal::SIGINT, SigHandler::SigDfl).unwrap(), SigHandler::SigIgn);
+
+        static mut RECEIVED_SIGINT: bool = false;
```

Comment:
`static mut` variables are `unsafe`.  The only reason the compiler allowed it here is because you put the entire test case into an `unsafe` block.  Why don't you use `unsafe` only where you need it?  I think the only reason that `signal` is `unsafe` is because of `mem::transmute`, right?

In any case, the safe way to do a static global is demonstrated by the `SIGNALED` variable in test/sys/test_aio.rs.

Reply:
Yes, the `mem::transmute` call + calling the `libc::signal` API. Should `signal` be marked safe?

Reply:
`signal` should only be marked safe if you can guarantee that the `mem::transmute` operation will always be safe.  In this case, I don't think you can, because `f` might be an `sa_sigaction` function previously set by a call to `sigaction`.